### PR TITLE
Subversion fixes

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: downloader/src/main/kotlin/vcs/Subversion.kt [hunk 1]
```diff
@@ -112,9 +114,13 @@ class Subversion : VersionControlSystem() {
         }
 
     override fun isApplicableUrlInternal(vcsUrl: String) =
-        runCatching {
-            clientManager.wcClient.doInfo(SVNURL.parseURIEncoded(vcsUrl), SVNRevision.HEAD, SVNRevision.HEAD)
-        }.isSuccess
+        try {
+            SVNRepositoryFactory.create(SVNURL.parseURIEncoded(vcsUrl)).checkPath("", -1) != SVNNodeKind.NONE
+        } catch (e: SVNException) {
+            log.debug { "An exception was thrown when checking $vcsUrl for a $type repository: $e" }
```
Comment (fviernau) [inline downloader/src/main/kotlin/vcs/Subversion.kt]: Consider putting $vcsUrl into quotes.
Would it make sense to log the full stack trace?

File: downloader/src/main/kotlin/VersionControlSystem.kt [hunk 2]
```diff
@@ -291,8 +291,10 @@ abstract class VersionControlSystem {
         }
 
         log.info {
-            "Successfully downloaded revision $workingTreeRevision which matches ${pkg.id.name} version " +
-                    "${pkg.id.version}."
+            val pkgName = " ${pkg.id.name}".trimEnd()
+            val pkgVersion = " version ${pkg.id.version}".takeUnless { pkg.id.version.isEmpty() }.orEmpty()
+            val whichMatches = " which matches".takeUnless { pkgName.isEmpty() && pkgVersion.isEmpty() }.orEmpty()
+            "Successfully downloaded revision $workingTreeRevision$whichMatches$pkgName$pkgVersion."
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply downloader/src/main/kotlin/VersionControlSystem.kt]: Done.

File: downloader/src/main/kotlin/vcs/Subversion.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Subversion.kt]: We usually do not quote URLs (as they cannot contain spaces, in contrast to e.g. file paths), so I also didn't do it here. But I'm calling `e.showStackTrace()` now.

File: downloader/src/main/kotlin/vcs/Subversion.kt [hunk 3]
```diff
@@ -112,9 +114,15 @@ class Subversion : VersionControlSystem() {
...
     override fun isApplicableUrlInternal(vcsUrl: String) =
-        runCatching {
-            clientManager.wcClient.doInfo(SVNURL.parseURIEncoded(vcsUrl), SVNRevision.HEAD, SVNRevision.HEAD)
-        }.isSuccess
+        try {
+            SVNRepositoryFactory.create(SVNURL.parseURIEncoded(vcsUrl)).checkPath("", -1) != SVNNodeKind.NONE
+        } catch (e: SVNException) {
+            e.showStackTrace()
+
+            log.debug { "An exception was thrown when checking $vcsUrl for a $type repository: $e" }
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Subversion.kt]: Use `${e.collectMessagesAsString}`.

Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Subversion.kt]: Done.

**Pull request merged**